src/validation/tests/property_validation_tests.rs

```diff
@@ -266,12 +398,269 @@ fn property_clashing_with_parent_property_name_is_ok() {
+        PROPERTY prop : STRING
+            GET END_GET
+        END_PROPERTY
+    END_FUNCTION_BLOCK
+    ";
+
+    insta::assert_snapshot!(test_utils::parse_and_validate_buffered(source), @r"
+    error[E112]: Overridden property `prop` has different signatures in POU `fb2` and `fb1`
+      ┌─ <internal>:9:18
+      │
+    3 │         PROPERTY prop : DINT
+      │                  ---- see also
+      ·
+    9 │         PROPERTY prop : STRING
+      │                  ^^^^ Overridden property `prop` has different signatures in POU `fb2` and `fb1`
+
+    error[E112]: Derived methods with conflicting signatures, return types do not match:
+      ┌─ <internal>:9:18
+      │
+    9 │         PROPERTY prop : STRING
+      │                  ^^^^ Derived methods with conflicting signatures, return types do not match:
+
+    note[E118]: Type `DINT` declared in `fb1.__get_prop` but `fb2.__get_prop` declared type `STRING`
+      ┌─ <internal>:3:18
+      │
+    3 │         PROPERTY prop : DINT
+      │                  ---- see also
+      ·
+    9 │         PROPERTY prop : STRING
+      │                  ---- see also
+    ");
+}
+
+#[test]
+fn overriding_property_in_interface_with_same_datatype_is_ok() {
+    let diagnostics = test_utils::parse_and_validate_buffered(
+        r"
+        INTERFACE intf1
+            PROPERTY prop : DINT
+                GET END_GET
+            END_PROPERTY
+        END_INTERFACE
+
+        INTERFACE intf2 EXTENDS intf1
+            PROPERTY prop : DINT
+                GET END_GET
+                SET END_SET
+            END_PROPERTY
+        END_INTERFACE
+        ",
+    );
+
+    // Essentially we're overriding the property in the child, which is OK because properties are methods
     insta::assert_snapshot!(diagnostics, @r"");
 }
 
 #[test]
-fn undefined_property_accessor_in_parent_yields_error() {
+fn overriding_property_in_interface_with_different_datatype_is_not_ok() {
+    let source = r"
+    INTERFACE intf1
+        PROPERTY prop : DINT
+            GET END_GET
+        END_PROPERTY
+    END_INTERFACE
+
+    // We extend the property by a SET accessor in this interface, but with the wrong datatype
+    INTERFACE intf2 EXTENDS intf1
+        PROPERTY prop : STRING 
+            SET END_SET
+        END_PROPERTY
+    END_INTERFACE
+    ";
+
+    insta::assert_snapshot!(test_utils::parse_and_validate_buffered(source), @r"
+    error[E112]: Property `prop` defined in interface `intf1` and `intf2` have different datatypes
+       ┌─ <internal>:9:15
+       │
+     3 │         PROPERTY prop : DINT
+       │                         ---- see also
+       ·
+     9 │     INTERFACE intf2 EXTENDS intf1
+       │               ^^^^^ Property `prop` defined in interface `intf1` and `intf2` have different datatypes
+    10 │         PROPERTY prop : STRING 
+       │                         ------ see also
+    ");
+}
+
+#[test]
+fn extending_interface_with_interfaces_with_conflicting_signatures_is_not_ok() {
```

Comment:
do we have a similar test for extending function blocks with conflicting property signatures?

Reply:
Kinda but I extended the coverage by two more tests